pkg/scheduler/scheduling_node.go

```diff
@@ -27,67 +27,158 @@ import (
 )
 
 type SchedulingNode struct {
-	NodeInfo *cache.NodeInfo
 	NodeId   string
 
-	// Resource which is allocating (in addition to confirmed, allocated)
-	AllocatingResource      *resources.Resource
-	PreemptingResource      *resources.Resource
-	CachedAvailableResource *resources.Resource
+	// Private info
+	nodeInfo           *cache.NodeInfo
+	allocatingResource *resources.Resource // resources being allocated
+	preemptingResource *resources.Resource // resources considered for preemption
+	cachedAvailable    *resources.Resource // calculated available resources
+	availableDirty     bool                // is the calculated available resource up to date?
```

Comment:
Is `cachedAvailableResourceExpires` a better name for this? Or any other alternative? `availableDirty` is not very self-explanatory to me. And also, this line deserves more comment to explain why we have this flag, which will help other to understand the idea.

Reply:
changed the name to `cachedAvailableUpToDate`